backend/src/main/java/org/example/backend/service/ShopService.java

```diff
@@ -23,4 +24,17 @@ public void addList(ShoppingListDto shoppingListDto) {
         ShoppingList shoppingList = new ShoppingList(idService.generateUUID(), shoppingListDto.listName(), new ArrayList<>());
         listRepo.save(shoppingList);
     }
+
+    public void updateList(String id, ShoppingListDto shoppingList) {
+        Optional<ShoppingList> foundList = listRepo.findById(id);
+        if (foundList.isPresent()) {
+            ShoppingList updateList = foundList.orElseThrow().withListName(shoppingList.listName());
```

Comment:
Du machst bereits einen Check in der Condition, daher brauchst du den .orElseThrow() nicht. Das hier würde ausreichen.
ShoppingList updateList = foundList.get().withListName(shoppingList.listName());

Reply:
Ich kann deinen Vorschlag nicht umsetzen, da die Funktion listRepo.findById(id) ein Optional voraussetzt.
Durch den Optional muss ein .orElseThrow gesetzt werden.